fdbserver/KeyValueStoreRocksDB.actor.cpp

```diff
@@ -271,6 +281,14 @@ struct RocksDBKeyValueStore : IKeyValueStore {
 				                          a.debugID.get().first(),
 				                          "Reader.Before"); //.detail("TaskID", g_network->getCurrentTask());
 			}
+			if (now() - a.startTime > SERVER_KNOBS->ROCKSDB_READ_VALUE_PREFIX_TIMEOUT) {
```

Comment:
One other note: you would have to run TSAN to confirm, but I'm pretty sure `now()` is thread hostile. In particular, it uses a global:

https://github.com/apple/foundationdb/blob/0452ff5759f0c2158aac9f9c3af9732920b99681/flow/flow.h#L1088

that is updated on the main thread:

https://github.com/apple/foundationdb/blob/0452ff5759f0c2158aac9f9c3af9732920b99681/flow/Net2.actor.cpp#L160
https://github.com/apple/foundationdb/blob/0452ff5759f0c2158aac9f9c3af9732920b99681/flow/Net2.actor.cpp#L191

Instead you should probably use `timer_monotic()` https://github.com/apple/foundationdb/blob/master/flow/Platform.h#L270